infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/Utils/CorfuSaasEndpointProvider.java

```diff
@@ -0,0 +1,57 @@
+package org.corfudb.infrastructure.logreplication.infrastructure.Utils;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Helper methods to retrieve the corfu's SaaS endpoint.
+ */
+@Slf4j
+public class CorfuSaasEndpointProvider {
+    private static Optional<CorfuSaasEndpointProvider> instance = Optional.empty();
```

Comment:
Codacy warning on final.

Reply:
will have to ignore codacy here, cannot make this variable final

Reply:
ohh! my bad...making the class final